src/main/java/dao/ProductDaoImpl.java

```diff
@@ -0,0 +1,33 @@
+package dao;
+
+import db.Storage;
+import java.util.List;
+import java.util.Objects;
+import model.Product;
+
+public class ProductDaoImpl implements ProductDao {
+
+    @Override
+    public Product get(String productName) {
+        return Storage.products.stream()
+                .filter(p -> Objects.equals(p.getName(), productName))
+                .findFirst().get();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        return Storage.products;
+    }
+
+    @Override
+    public void update(Product product) {
+        Product productToRemove = get(product.getName());
+        Storage.products.remove(productToRemove);
```

Comment:
for this operation, it's better to override equals in Product because otherwise, you can have a situation when needed fruit won't be removed